pandas/core/dtypes/cast.py

```diff
@@ -357,12 +357,18 @@ def maybe_cast_result_dtype(dtype: DtypeObj, how: str) -> DtypeObj:
         The desired dtype of the result.
     """
     from pandas.core.arrays.boolean import BooleanDtype
-    from pandas.core.arrays.integer import Int64Dtype
-
-    if how in ["add", "cumsum", "sum"] and (dtype == np.dtype(bool)):
-        return np.dtype(np.int64)
-    elif how in ["add", "cumsum", "sum"] and isinstance(dtype, BooleanDtype):
-        return Int64Dtype()
+    from pandas.core.arrays.floating import Float64Dtype
+    from pandas.core.arrays.integer import Int64Dtype, _IntegerDtype
```

Comment:
i think the linter is going to complain about _IntegerDtype.   we can either find a non-private thing to import or add it to the whitelist in scripts._validate_unwanted_patterns

Reply:
> i think the linter is going to complain about _IntegerDtype

Apparantly it's not complaining at the moment, but indeed something we can de-privatize internally